std/rb.zig

```diff
@@ -0,0 +1,523 @@
+            }
+        }
+        // This was an insert, there is at least one node.
+        tree.root.?.set_color(Black);
+        return null;
+    }
+
+    pub fn lookup(tree: *Tree, key: *Node) ?*Node {
+        var parent: *Node = undefined;
+        var is_left: bool = undefined;
+
+        return do_lookup(key, tree, &parent, &is_left);
+    }
+
+    pub fn remove(tree: *Tree, nodeconst: *Node) void {
+        var node = nodeconst;
+        // as this has the same value as node, it is unsafe to access node after newnode
+        var newnode: ?*Node = nodeconst;
+        var maybe_parent: ?*Node = node.get_parent();
+        var color: Color = undefined;
+        var next: *Node = undefined;
+
+        // This clause is to avoid optionals
+        if (node.left == null and node.right == null) {
+            if (maybe_parent) |parent| {
+                parent.set_child(null, parent.left == node);
+            } else
+                tree.root = null;
+            color = node.get_color();
+            newnode = null;
+        } else {
+            if (node.left == null) {
+                next = node.right.?; // Not both null as per above
+            } else if (node.right == null) {
+                next = node.left.?;  // Not both null as per above
+            } else
+                next = node.right.?.get_first(); // Just checked for null above
+
+            if (maybe_parent) |parent| {
+                parent.set_child(next, parent.left == node);
+            } else
+                tree.root = next;
+
+            if (node.left != null and node.right != null) {
+                if (node.left) |left| {
+                    if (node.right) |right| {
+                        color = next.get_color();
+                        next.set_color(node.get_color());
+
+                        next.left = left;
+                        left.set_parent(next);
+
+                        if (next != right) {
+                            var parent = next.get_parent().?; // Was traversed via child node (right/left)
+                            next.set_parent(node.get_parent());
+
+                            newnode = next.right;
+                            parent.left = node;
+
+                            next.right = right;
+                            right.set_parent(next);
+                        } else {
+                            next.set_parent(maybe_parent);
+                            maybe_parent = next;
+                            newnode = next.right;
+                        }
+                    }
+                }
+            } else {
+                color = node.get_color();
+                newnode = next;
+            }
+        }
+
+        if (newnode) |n|
+            n.set_parent(maybe_parent);
+
+        if (color == Red)
+            return;
+        if (newnode) |n| {
+            n.set_color(Black);
+            return;
+        }
+
+        while (true) {
+            if (node == tree.root)
```

Comment:
This if-break statement is at the top of `while (true)` which means you should just make it the while condition.